subprojects/language-java/src/testFixtures/groovy/org/gradle/language/fixtures/ServiceRegistryProcessorFixture.groovy

```diff
@@ -23,62 +23,59 @@ import javax.tools.StandardLocation
 
 /**
  * A processor that collects all types annotated with the @Service annotation
- * and generates a single ServiceRegistry class from them. The registry has
- * one getter for each annotated type, which calls the no-argument constructor
- * of that type.
+ * and generates a single ServiceRegistryResource.txt file from them.
+ * Also creates a class called ServiceRegistry which does not reference any of the
+ * other classes, but could simply serve as an entry point for reading the resource file.
  */
 @CompileStatic
 class ServiceRegistryProcessorFixture extends AnnotationProcessorFixture {
```

Comment:
This is now much closer to what Micronaut does and will hopefully prevent future regressions. Also see the commit message for why this no longer creates an aggregating class file.

Reply:
I'll try what happens with a class file, would be worth the additional coverage. Micronaut writes a [resource file](https://github.com/micronaut-projects/micronaut-core/blob/a949a27c69ff00fe50fa751adf0501540aee79b2/inject/src/main/java/io/micronaut/inject/writer/AbstractClassWriterOutputVisitor.java#L99).

Reply:
Same happens with a class file, which makes sense since class files can be annotation-processed. You get a warning that a class was generated in the last round and no other code can reference that class.